scheduler/reconciler/reconcile_cluster_prop_test.go

```diff
@@ -0,0 +1,399 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package reconciler
+
+import (
+	"encoding/binary"
+	"fmt"
+	"maps"
+	"slices"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/nomad/helper"
+	"github.com/hashicorp/nomad/helper/pointer"
+	"github.com/hashicorp/nomad/helper/testlog"
+	"github.com/hashicorp/nomad/nomad/structs"
+	"pgregory.net/rapid"
+)
+
+const maxAllocs = 30
```

Comment:
Is there a scenario where we would want to be able to configure/alter this?

Reply:
Maybe, but I don't have one in mind at the moment. Right now there's only one sub-generator that uses this, and that's for the task group count and is a range from 0 to `maxAllocs`.